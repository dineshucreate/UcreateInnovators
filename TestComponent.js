import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

class TestComponent extends React.Component {

    state = {
        result: 0
    }

    componentDidMount() {
        console.log('...........componentDidMount..............');

    }


    sum = (a, b) => {
        return a + b;
    }

    render() {
        console.log('...........render..............');

        return (
            <View style={{ flex: 1 }}>
                <Text>Sum Value is {this.state.result}</Text>
            </View>
        );
    }
}
export default TestComponent;
