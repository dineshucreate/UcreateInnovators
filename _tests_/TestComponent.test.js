import React from 'react';
import TestComponent from '../TestComponent';
import { shallow } from 'enzyme';
import 'react-native';
import 'react-test-renderer';
import toJson from 'enzyme-to-json';

describe('App Snapshot', () => {
    const app = shallow(<TestComponent />)
    it('rendering ', () => {
        expect(toJson(app)).toMatchSnapshot();
    });

});

describe('sum suite', () => {
    test('Should add 2 positive numbers together and return the result', () => {
        const wrapper = shallow(<TestComponent />).instance();
        expect(wrapper.sum(2, 3)).toBe(5);
    });
});