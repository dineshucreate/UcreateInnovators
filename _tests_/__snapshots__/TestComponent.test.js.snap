// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot rendering  1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text>
    Sum Value is 
    0
  </Text>
</View>
`;
